packages/backend/src/models/PasswordResetLinkModel.ts

```diff
@@ -0,0 +1,93 @@
+import { PasswordResetLink } from 'common';
+import * as crypto from 'crypto';
+import { Knex } from 'knex';
+import { URL } from 'url';
+import { LightdashConfig } from '../config/parseConfig';
+import { EmailTableName } from '../database/entities/emails';
+import { PasswordResetTableName } from '../database/entities/passwordResetLinks';
+import { NotExistsError } from '../errors';
+
+type Dependencies = {
+    database: Knex;
+    lightdashConfig: LightdashConfig;
+};
+
+export class PasswordResetLinkModel {
+    private database: Knex;
+
+    private lightdashConfig: LightdashConfig;
+
+    constructor({ database, lightdashConfig }: Dependencies) {
+        this.database = database;
+        this.lightdashConfig = lightdashConfig;
+    }
+
+    transformCodeToUrl(code: string): string {
+        return new URL(`/reset-password/${code}`, this.lightdashConfig.siteUrl)
+            .href;
+    }
+
+    static _hash(s: string): string {
+        return crypto.createHash('sha256').update(s).digest('hex');
+    }
+
+    async getByCode(code: string): Promise<PasswordResetLink> {
+        const codeHash = PasswordResetLinkModel._hash(code);
+        const links = await this.database(PasswordResetTableName)
+            .leftJoin(
+                EmailTableName,
+                'emails.email_id',
+                'password_reset_links.email_id',
+            )
+            .select<{ expires_at: Date; email: string }[]>([
+                'password_reset_links.expires_at',
+                'emails.email',
+            ])
+            .where('code_hash', codeHash);
+        if (links.length === 0) {
+            throw new NotExistsError('No password reset link found');
+        }
+        const passwordResetLink = links[0];
+        return {
+            code,
+            email: passwordResetLink.email,
+            expiresAt: passwordResetLink.expires_at,
+            url: this.transformCodeToUrl(code),
+        };
+    }
+
+    async create(
+        code: string,
+        expiresAt: Date,
+        email: string,
+    ): Promise<PasswordResetLink> {
+        const codeHash = PasswordResetLinkModel._hash(code);
+        const emails = await this.database(EmailTableName)
```

Comment:
I don't think you need this check since inserting into password reset table requires a foreign key to the email. So the insert will fail if the email can't be found

Reply:
This isn't just a check. I'm fetching the `email_id` for the `email`.